clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -98,17 +114,12 @@ mod clarity_v2_v3 {
             );
 
             let expected = match version_byte {
-                // Valid range for version_bytes
-                0x00..=0x1f => {
-                    match version_byte {
-                        // Since tests runs on a Testnet version,
-                        // version_bytes single_sig (0x1A) || multi_sig (0x15), for Testnet,
-                        // will return an Ok value.
-                        0x1A | 0x15 => Value::okay(expected_principal),
-                        _ => Ok(create_error_construct(0, Some(expected_principal))),
-                    }
-                },
-                _ => Ok(create_error_construct(1, None)),
+                 // Since tests runs on a Testnet version,
+                 // version_bytes single_sig (0x1A) || multi_sig (0x15), for Testnet,
+                 // will return an Ok value.
+                0x1A | 0x15 => Value::okay(expected_principal),
+                0x00..=0x1F => Ok(create_error_construct(0, Some(expected_principal))),
+                _ => Ok(create_error_construct(1, None))
```

Comment:
As `strategies_for_version_byte()` now returns only bytes in `0x00..=0x1F`, is this catchall case still possible? If not, can we change it to some `unreachable`/`panic`/something else, to make it clear we are not testing that case? Also for `crosscheck_principal_destruct`

Reply:
Yep, let's `panic!` to make sure that the catchall match arm is not tested.